06_puppy/mohitnag/memstore.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import "fmt"
+
+// CreatePuppy creates a Puppy in memstore
+func (m *MemStore) CreatePuppy(p Puppy) error {
+	if _, ok := (*m)[p.ID]; !ok {
+		(*m)[p.ID] = p
+		return nil
+	}
+	return fmt.Errorf("puppy with Id %d already exists", p.ID)
+}
+
+// ReadPuppy reads a Puppy from memstore
+func (m *MemStore) ReadPuppy(id uint32) (Puppy, error) {
+	if _, ok := (*m)[id]; !ok {
+		return Puppy{}, fmt.Errorf("puppy with Id %d does not exists", id)
+	}
+	return (*m)[id], nil
+}
+
+// UpdatePuppy updates a Puppy
+func (m *MemStore) UpdatePuppy(p Puppy) error {
+	if _, ok := (*m)[p.ID]; !ok {
+		return fmt.Errorf("puppy with Id %d does not exists", p.ID)
+	}
+	(*m)[p.ID] = p
+	return nil
+}
+
+// DeletePuppy deletes a Puppy
+func (m *MemStore) DeletePuppy(id uint32) bool {
```

Comment:
better use error as return val than bool. if your datastore is backed by a database or similar any operation can error (connection timeout etc.).
needs to be fixed in interface as well.

Reply:
Done